addons/l10n_it_edi_ndd/models/account_move.py

```diff
@@ -22,12 +22,16 @@ class AccountMove(models.Model):
     @api.depends('line_ids.matching_number', 'payment_state')
     def _compute_l10n_it_payment_method(self):
         move_lines_per_matching_number = self.env['account.move.line'].search([
-            ('matching_number', 'in', self.line_ids.filtered('matching_number').mapped('matching_number'))
+            ('matching_number', 'in', self.line_ids.filtered('matching_number').mapped('matching_number')),
+            ('company_id.account_fiscal_country_id.code', '=', 'IT'),
```

Comment:
```suggestion
            ('company_id', '=', self.env.company.id),
```
I think we could even restrict the domain further to only include lines from the current company (since we know it will be Italian by now).